src/fft/fftn.rs

```diff
@@ -0,0 +1,534 @@
+    let num_s = s.len();
+    let num_axes = axes.len();
+
+    let s_ptr = s.as_ptr();
+    let axes_ptr = axes.as_ptr();
+
+    unsafe {
+        let c_array =
+            mlx_sys::mlx_fft_ifftn(a.c_array, s_ptr, num_s, axes_ptr, num_axes, stream.as_ptr());
+        Array::from_ptr(c_array)
+    }
+}
+
+/// n-dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `s`: Sizes of the transformed axes. The corresponding axes in the input are truncated or
+///  padded with zeros to match the sizes in `s`. The default value is the sizes of `a` along `axes`
+///  if not specified.
+/// - `axes`: Axes along which to perform the FFT. The default is `None` in which case the FFT is
+///   over the last `len(s)` axes are or all axes if `s` is also `None`.
+#[default_device(device = "cpu")]
+pub unsafe fn ifftn_device_unchecked<'a>(
+    a: &'a Array,
+    s: impl Into<Option<&'a [i32]>>,
+    axes: impl Into<Option<&'a [i32]>>,
+    stream: StreamOrDevice,
+) -> Array {
+    let (valid_s, valid_axes) = resolve_sizes_and_axes_unchecked(a, s, axes);
+    ifftn_device_inner(a, &valid_s, &valid_axes, stream)
+}
+
+/// n-dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `s`: Sizes of the transformed axes. The corresponding axes in the input are truncated or
+/// padded with zeros to match the sizes in `s`. The default value is the sizes of `a` along `axes`
+/// if not specified.
+/// - `axes`: Axes along which to perform the FFT. The default is `None` in which case the FFT is
+/// over the last `len(s)` axes are or all axes if `s` is also `None`.
+#[default_device(device = "cpu")]
+pub fn try_ifftn_device<'a>(
+    a: &'a Array,
+    s: impl Into<Option<&'a [i32]>>,
+    axes: impl Into<Option<&'a [i32]>>,
+    stream: StreamOrDevice,
+) -> Result<Array, FftError> {
+    let (valid_s, valid_axes) = try_resolve_sizes_and_axes(a, s, axes)?;
+    Ok(ifftn_device_inner(a, &valid_s, &valid_axes, stream))
+}
+
+/// n-dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `s`: Sizes of the transformed axes. The corresponding axes in the input are truncated or
+/// padded with zeros to match the sizes in `s`. The default value is the sizes of `a` along `axes`
+/// if not specified.
+/// - `axes`: Axes along which to perform the FFT. The default is `None` in which case the FFT is
+/// over the last `len(s)` axes are or all axes if `s` is also `None`.
+///
+/// # Panic
+///
+/// Panics if the input arguments are invalid. See [try_ifftn_device] for more details.
+#[default_device(device = "cpu")]
+pub fn ifftn_device<'a>(
+    a: &'a Array,
+    s: impl Into<Option<&'a [i32]>>,
+    axes: impl Into<Option<&'a [i32]>>,
+    stream: StreamOrDevice,
+) -> Array {
+    try_ifftn_device(a, s, axes, stream).unwrap()
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I like to add tests for known failures -- but if you think its enough to be covered by the tests in the utils you're using to validate, then I leave it up to you.

Reply:
I think this would suffice